### ISSUE TITLE
node_modules/expo/AppEntry.js: [BABEL] /node_modules/expo/AppEntry.js: Unknown option: .name.

### ISSUE THREAD

**Issue by Arowne** [expo/expo#10616]
## 🐛 Bug Report
node_modules/expo/AppEntry.js: [BABEL] /node_modules/expo/AppEntry.js: Unknown option: .name. Check out https://babeljs.io/docs/en/babel-core/#options for more information about options.

### Summary of Issue
 i'm facing this use with babel with in the expo folder in node_modules this  file only contain an import of App object in a TS object (I don't easily see where babel is in this file) 

### Environment 
  Expo CLI 3.27.14 environment info:
    System:
      OS: Linux 5.4 Ubuntu 18.04.5 LTS (Bionic Beaver)
      Shell: 4.4.20 - /bin/bash
    Binaries:
      Node: 14.0.0 - /usr/local/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 6.14.4 - /usr/local/bin/npm
    npmPackages:
      expo: 39.0.0 => 39.0.0 
      react: ^16.13.1 => 16.13.1 
      react-dom: ^16.13.1 => 16.13.1 
      react-native: ^0.63.3 => 0.63.3 
      react-native-web: ^0.12.2 => 0.12.3 
      react-navigation: ^4.3.9 => 4.4.2 
    npmGlobalPackages:
      expo-cli: 3.27.14
    Expo Workflow: managed

### Steps to Reproduce
Here my package.json:
`{
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "@azure/cognitiveservices-customvision-prediction": "^5.1.2",
    "@azure/cognitiveservices-customvision-training": "^5.1.2",
    "@azure/ms-rest-js": "^2.0.8",
    "@expo/vector-icons": "^10.1.0",
    "@freakycoder/react-native-header-view": "^0.5.1",
    "@freakycoder/react-native-helpers": "^0.1.3",
    "@google-cloud/translate": "^5.3.0",
    "@react-native-community/art": "^1.2.0",
    "@react-native-community/masked-view": "^0.1.6",
    "@react-native-community/viewpager": "^4.0.0",
    "@tensorflow/tfjs": "^2.3.0",
    "@tensorflow/tfjs-react-native": "^0.3.0",
    "aws-sdk": "^2.733.0",
    "axios": "^0.19.2",
    "babel-plugin-react-transform": "^3.0.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-expo": "^8.3.0",
    "babel-preset-react": "^6.24.1",
    "babel-preset-react-native": "^4.0.1",
    "bootstrap": "^4.4.1",
    "capitalize": "^2.0.3",
    "clarifai": "^2.9.1",
    "customvision-api": "0.0.2",
    "ex-react-native-i18n": "0.0.6",
    "expo": "39.0.0",
    "expo-ads-admob": "^8.1.0",
    "expo-av": "^8.1.0",
    "expo-barcode-scanner": "^8.1.0",
    "expo-camera": "^8.1.0",
    "expo-cli": "^3.20.1",
    "expo-font": "^8.1.1",
    "expo-gl": "^8.4.0",
    "expo-google-app-auth": "^8.1.0",
    "expo-image-picker": "^8.1.0",
    "expo-linear-gradient": "^8.2.1",
    "expo-localization": "^8.2.1",
    "expo-sensors": "^8.1.0",
    "expo-speech": "^8.1.0",
    "expo-vector-icons": "^10.0.1",
    "expokit": "^37.0.0",
    "express-form": "^0.12.6",
    "form-data": "^3.0.0",
    "fuse.js": "^6.4.1",
    "galio-framework": "^0.6.3",
    "i": "^0.3.6",
    "i18n-js": "^3.7.1",
    "install": "^0.13.0",
    "jest": "^26.0.1",
    "jetifier": "^1.6.4",
    "match-sorter": "^4.1.0",
    "metro-react-native-babel-preset": "0.56.0",
    "nachos-ui": "^0.2.0-beta.1",
    "npm": "^6.14.8",
    "package.json": "^2.0.1",
    "pluralize": "^8.0.0",
    "react": "^16.13.1",
    "react-content-loader": "^5.1.0",
    "react-dom": "^16.13.1",
    "react-loader-spinner": "^3.1.14",
    "react-native": "^0.63.3",
    "react-native-action-button": "^2.8.5",
    "react-native-admob": "^1.3.2",
    "react-native-androw": "0.0.34",
    "react-native-animatable": "^1.3.3",
    "react-native-app-intro-slider": "^4.0.4",
    "react-native-autocomplete-input": "^4.1.0",
    "react-native-barcode-mask": "^1.2.4",
    "react-native-bouncing-preloader": "^1.0.1",
    "react-native-dotenv": "^2.4.1",
    "react-native-dropdown-autocomplete": "^1.0.18",
    "react-native-dropdown-menu": "^2.0.0",
    "react-native-dropdown-picker": "^2.1.1",
    "react-native-dynamic-vector-icons": "^1.0.0",
    "react-native-easy-content-loader": "^0.3.2",
    "react-native-elements": "^2.0.0",
    "react-native-file-picker": "0.0.19",
    "react-native-file-upload": "^1.0.4",
    "react-native-fs": "^2.16.6",
    "react-native-gaid": "^1.0.0",
    "react-native-gesture-handler": "^1.6.1",
    "react-native-header-search-bar": "^0.1.10",
    "react-native-idfa": "^4.1.0",
    "react-native-image-picker": "^2.3.1",
    "react-native-image-to-blob": "0.0.6",
    "react-native-indicator": "^1.2.2",
    "react-native-linear-gradient": "^2.5.6",
    "react-native-loader": "^1.2.1",
    "react-native-material-cards": "^1.0.15",
    "react-native-material-dropdown": "^0.11.1",
    "react-native-modal-selector": "^2.0.3",
    "react-native-navigation": "^6.7.5",
    "react-native-navigation-search-bar": "^1.0.6",
    "react-native-paper": "^3.9.0",
    "react-native-picker-select": "^8.0.0",
    "react-native-progressive-input": "^1.1.0",
    "react-native-reanimated": "^1.7.0",
    "react-native-restart": "0.0.17",
    "react-native-safe-area-context": "^0.7.3",
    "react-native-screens": "^2.2.0",
    "react-native-select-input-ios": "^2.0.5",
    "react-native-shake-event": "^4.0.1",
    "react-native-shimmer-placeholder": "^1.0.36",
    "react-native-snap-carousel": "^3.9.0",
    "react-native-svg": "^12.1.0",
    "react-native-svg-animated-linear-gradient": "^0.4.0",
    "react-native-swiper": "^1.6.0",
    "react-native-touchable-scale": "^2.1.1",
    "react-native-unimodules": "^0.9.1",
    "react-native-vector-icons": "^6.6.0",
    "react-native-video": "^4.4.5",
    "react-native-web": "^0.12.2",
    "react-navigation": "^4.3.9",
    "react-navigation-material-bottom-tabs": "^2.2.12",
    "react-navigation-stack": "^2.4.0",
    "react-navigation-tabs": "^2.8.13",
    "react-test-renderer": "^16.13.1",
    "reactstrap": "^8.4.1",
    "rn-content-loader": "0.0.6",
    "rng": "^0.2.2",
    "scheduler": "^0.19.1",
    "shortid": "^2.2.15",
    "shuffle-array": "^1.0.1",
    "socket.io-client": "^2.3.0",
    "styled-components": "^5.1.0",
    "sync-storage": "^0.4.2",
    "unsplash-js": "^6.0.0",
    "uuid": "^7.0.3",
    "watchman": "^1.0.0",
    "yarn": "^1.22.4",
    "yarn.lock": "^0.0.1-security",
    "zeplin-to-react-native": "^1.0.0"
  },
  "devDependencies": {
    "@babel/core": "^7.8.6",
    "babel-core": "^7.0.0-bridge.0",
    "babel-plugin-module-resolver": "^4.0.0",
    "babel-plugin-react-native-platform-specific-extensions": "^1.1.1"
  },
  "private": true,
  "name": "RecettesFaciles",
  "version": "1.0",
  "license": "MIT"
}`

### Expected Behavior vs Actual Behavior
The project not run at all.

**Comment by byCedric** [comment]
Hi @Arowne, I'm sorry you are having issues with this. Could you [provide a repo](https://stackoverflow.com/help/minimal-reproducible-example) because I can't seem to reproduce this issue on my end. With that, we can dig into this and solve this issue for you!

**Comment by dpickut** [comment]
I've run into this problem also.  Using react-native @39.0.3.  

I've looked at node_modules/expo/AppEntry.js -- it is a very simple file, and the path to App.js is correct. There is no ".name" option specified. 

Here's the AppEntry.js code:
import registerRootComponent from 'expo/build/launch/registerRootComponent';
import App from '../../App';
registerRootComponent(App);

**Comment by brentvatne** [comment]
hi! we need a reproducible example to help with this. please create a new issue and provide an example that we can run to demonstrate the issue. this is likely a bug in your own code and trying to provide a minimal reproducible example will help you to locate that bug! if you can reproduce it then it'll help us to investigate! more guidance on how to do this if you need it: https://stackoverflow.com/help/minimal-reproducible-example

**Comment by laudeon** [comment]
Hello,

I have the exact same issue, using sdk 39.0.3. 

I figured out that if I remove the necessary babel config related to the `react-native-dotenv` module : `'module:react-native-dotenv'` in the presets list, then I got this error message : `The package at "node_modules/react-native-dotenv/index.js" attempted to import the Node standard library module "fs". It failed because the native React runtime does not include the Node standard library.`. 

So I just get ride of this deps in my codebase (debugging purpose, I want to be able to use it) and everything works fine.
So my guess is that the bug might be an issue with this specific package and not related to expo.
